src/compile/model.ts

```diff
@@ -182,8 +182,8 @@ export abstract class Model {
     this.parseScale();
     this.parseMarkDef();
     this.parseLayoutSize(); // depends on scale
-    this.parseData(); // (pathorder) depends on markDef
     this.parseSelection();
+    this.parseData(); // (pathorder) depends on markDef
```

Comment:
Sorry for a delayed comment. 

@arvind  It seems like this comment is not addressed yet?  (At least for the merged commit in the master branch)